agent/src/storage/blobs.rs

```diff
@@ -130,12 +86,56 @@ impl BlobStore {
             Err(_) => Ok(false), // TODO: probably should handle errors more granularly
         }
     }
+}
+
+#[async_trait]
+impl RunnerStorage for &BlobStore {
+    /// Fetch a manifest by its fully-qualified name.
+    async fn manifest(&self, fq_name: &str) -> Result<Manifest, ServalError> {
+        let bytes = cacache::read(&self.location, Manifest::make_manifest_key(fq_name)).await?;
+        if let Ok(data) = String::from_utf8(bytes) {
+            let manifest: Manifest = toml::from_str(&data)?; // TODO CJ this is json right now
+            Ok(manifest)
+        } else {
+            // TODO: bad data error
+            Err(ServalError::ManifestNotFound(fq_name.to_string()))
+        }
+    }
+
+    /// A non-streaming way to retrieve a stored compiled Wasm task. Prefer executable_as_stream() if you do not
+    /// need the executable bytes in memory.
+    async fn executable_as_bytes(&self, name: &str, version: &str) -> Result<Vec<u8>, ServalError> {
+        let key = Manifest::make_executable_key(name, version);
+        let binary: Vec<u8> = cacache::read(&self.location, key).await?;
+        Ok(binary)
+    }
 
-    pub fn manifest_names(&self) -> Result<Vec<String>, ServalError> {
+    /// Retrieve a list of all Wasm manifests stored on this node.
+    async fn manifest_names(&self) -> Result<Vec<String>, ServalError> {
         let result: Vec<String> = cacache::list_sync(&self.location)
             .filter(|xs| xs.is_ok())
             .map(|xs| xs.unwrap().key)
+            .filter(|xs| xs.contains("manifest"))
```

Comment:
(I am just riffing because coding is fun)

We _could_ smoosh this into a single `.filter_map` rather than `.filter.map.filter`:

````rs
let result: Vec<String> = cacache::list_sync(&self.location)
            .filter_map(|xs| match xs {
                Ok(xs) if xs.key.contains("manifest") => Some(xs.key),
                _ => None,
            })
            .collect();
````

Whether this is a grokkability improvement remains to be seen; I kinda like it more but it's not a slam dunk.